gravitee-am-service/src/test/java/io/gravitee/am/service/validators/annotations/IfNotNullThenNotBlankValidatorTest.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.service.validators.annotations;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class IfNotNullThenNotBlankValidatorTest {
+
+    private IfNotNullThenNotBlankValidator validator;
+    @BeforeAll
+    void setUp() {
+        validator = new IfNotNullThenNotBlankValidator();
+    }
+
+    @ParameterizedTest
+    @MethodSource("validValues")
+    void isValid(String value) {
+        assertTrue(validator.isValid(value, null));
+    }
+
+    public Stream<String> validValues() {
+        return Stream.of("blah", "_", "  test  ", null);
+    }
+    @ParameterizedTest
+    @MethodSource("invalidValues")
```

Comment:
I think that this should do the trick, sparing us the `invalidValues` method
```suggestion
    @ValueSource(strings = {"", "   "})
```